src/pages/MainPage/MainPage.js

```diff
@@ -1,9 +1,230 @@
 import React, { Component } from 'react';
+import MainItem from './MainItem/MainItem';
 import './MainPage.scss';
 
 class MainPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      eventCount: 1,
+      insideCount: 1,
+      list: [],
+    };
+    this.saleMargin = 0;
+    this.insideMargin = 0;
+  }
+
+  handleNextClick = () => {
+    const { eventCount, list } = this.state;
+    const length = Number(list.map(item => item.eventList.length));
+    console.log(length);
+
+    if (eventCount < length) {
+      this.setState({
+        eventCount: eventCount + 1,
+      });
+      this.saleMargin -= 421;
+    }
+  };
+
+  handlePrevClick = () => {
+    const { eventCount, list } = this.state;
+    const length = Number(list.map(item => item.eventList.length));
+
+    if (!eventCount < length && eventCount > 1) {
+      this.setState({
+        eventCount: eventCount - 1,
+      });
+      this.saleMargin += 421;
+    }
+  };
+
+  handleInSideNextClick = () => {
+    const { insideCount, list } = this.state;
+    const length = Number(list.map(item => item.insideList.length));
+
+    if (insideCount < length) {
+      this.setState({
+        insideCount: insideCount + 1,
+      });
+
+      this.insideMargin -= 1562;
+    }
+  };
+
+  handleInSidePrevClick = () => {
+    const { insideCount, list } = this.state;
+    const length = Number(list.map(item => item.insideList.length));
+
+    if (!insideCount < length && insideCount > 1) {
+      this.setState({
+        insideCount: insideCount - 1,
+      });
+
+      this.insideMargin += 1562;
+    }
+  };
+
+  componentDidMount() {
+    fetch('http://localhost:3007/data/mainData-songhyun.json')
+      .then(res => res.json())
+      .then(lists => {
+        const init = {
+          insideList: [],
+          eventList: [],
+          collectionList: [],
+        };
+
+        lists.forEach(item => {
+          const obj = {
+            inside: 'insideList',
+            collection: 'collectionList',
+            event: 'eventList',
+          };
+          init[obj[item.category]].push(item);
+        });
+        this.setState({ list: [init] });
```

Comment:
```suggestion
        this.setState({
          insideList: [...],
          eventList: [...],
          collectionList: [...],
        });
```
이런 형태로 될 수 있도록 다 만들어두셨잖아요...ㅠㅠ state에 있는 list를 3종류의 state로 분리하라는 뜻이었습니다

Reply:
뜨끔....종택님 새벽에 안 주무셨어요? 리뷰반영하여 수정했습니다!

Reply:
반영하신 것 push 안된 것 같네요